MonstieWash/Assets/Scripts/Traversal/TraversalObject.cs

```diff
@@ -6,15 +6,35 @@ public class TraversalObject : MonoBehaviour, INavigator
     [SerializeField] private bool targetIsUI;
 
     private string m_targetScene;
+    private bool m_traversalEnabled = true;
+    private SpriteRenderer m_spriteRenderer;
 
     public void Awake()
     {
         if (targetScene == null) Debug.LogError($"Target scene not assigned for {name}!");
         else m_targetScene = targetScene.SceneName;
+
+        var monsterController = FindFirstObjectByType<MonsterController>();
+        monsterController.OnAttackBegin += MonsterController_OnAttackStart;
+        monsterController.OnAttackEnd += MonsterController_OnAttackEnd;
+
+        m_spriteRenderer = GetComponent<SpriteRenderer>();
+    }
+
+    private void MonsterController_OnAttackEnd()
+    {
+        m_traversalEnabled = true;
+        m_spriteRenderer.color = Color.white;
+    }
+
+    private void MonsterController_OnAttackStart()
+    {
+        m_traversalEnabled = false;
+        m_spriteRenderer.color = new Color(0.5f, 0f, 0f);
```

Comment:
Might be better to save the color of the sprite on awake and then use it here instead of the hard coded value in case designers want to change them later.